numpy/core/src/multiarray/ctors.c

```diff
@@ -1633,7 +1633,7 @@ _array_from_buffer_3118(PyObject *memoryview)
  * a NUMPY_API function, we cannot remove it.
  */
 NPY_NO_EXPORT int
-PyArray_GetArrayParamsFromObject(PyObject *op,
+PyArray_GetArrayParamsFromObject_int(PyObject *op,
```

Comment:
Can you drop the NULL context argument?

Reply:
Hmmm, although, then I need to move the error to the `PyArray_GetArrayParamsFromObject`. But I guess that is fine.

Reply:
OK, fixed this, and expanded Erics comment. I think I will look into array coercion replacement (with the idea that I can write this so that when we get new DTypes the code changes there will be small).

Hopefully, that will likely include simply moving this whole function onto a "will be deleted later" pile.